volmdlr/wires.py

```diff
@@ -1123,51 +1125,50 @@ def is_sharing_primitives_with(self, contour, all_points=False):
                         if list_p != [] and point.point_distance(point.nearest_point(list_p)) > 1e-4:
                             list_p.append(point)
 
-                        # edges2.add(edge2)
+                    if len(list_p) == 2:
+                        return True
+        return False
+
+    def shared_primitives_extremities(self, contour):
+        '''
+        extract shared primitives extremities between two adjacent contours
+        '''
+
+        if self.is_overlapping(contour):
+            raise ValueError('The contours are overlapping')
+        if self.is_superposing(contour):
+            raise ValueError('The contours are superposing')
```

Comment:
If the contours are superposing or overlapping, does it mean they do not share any primitives? 
if this is the case, shouldn't we return []? Or is it really a problem and we have to stop de code?

Reply:
Actually, I was thinking about this yesterday ! :p 
Two contours may be overlapping, but are sharing primitives, like here:
![image](https://user-images.githubusercontent.com/68638483/167882352-7767564b-97cc-4178-876e-b236cffc5d8d.png)